app/data_model/answer_store.py

```diff
@@ -187,6 +188,8 @@ def filter(self, group_id=None, block_id=None, answer_id=None, group_instance=No
                     matches = matches and answer[k] == v
             if matches:
                 filtered.append(answer)
+                if limit and len(filtered) == limit:
+                    break
```

Comment:
Not sure I always prefer to be explicit in what I want to happen i.e. return the list when it's met 25 but maybe a preference thing here so I'm happy to let go